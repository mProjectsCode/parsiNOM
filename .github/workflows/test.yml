name: Test

on:
  push:
    branches: [master, main]
  pull_request:
    branches: [master, main]

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: oven-sh/setup-bun@v1
        with:
          bun-version: latest
      - working-directory: .
        run: bun install
      - working-directory: .
        run: bun run check
