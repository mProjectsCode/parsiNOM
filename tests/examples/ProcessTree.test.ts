import { describe } from 'bun:test';
import { testParser } from '../TestHelpers';
import { ProcessTreeParser } from '../../examples/ProcessTree';

describe.each([
	// basic trees
	['a', true],
	['\\tau', true],
	['\\rightarrow(a)', true],
	['\\rightarrow(\\tau)', true],

	// smoke tests - a few process trees
	['\\rightarrow(a,\\circlearrowleft(\\rightarrow(\\wedge(\\times(b, c), d), e ), f ), \\times(g, h))', true],

	// Incorrect trees: empty tree
	['', false],
	// Numbers are not valid event names
	['1', false],
	// Incomplete/typoed sequence operation
	['\\rightarrow', false],
	['\\rightarrow(', false],
	['\\rightarrow)', false],
	['\\rightarrow(a', false],
	['\\rightarrowa)', false],
	// Disallowed starting operations
	['\\wedge(a)', false],
	['\\circlearrowleft(a)', false],
	['\\times(a)', false],
])(`Process Tree Testsuite`, (str, shouldSucceed) => {
	testParser(ProcessTreeParser, str, shouldSucceed);
});
