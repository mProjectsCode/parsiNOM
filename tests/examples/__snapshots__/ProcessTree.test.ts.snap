// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Process Tree Testsuite 'a' conditional success tests AST matches snapshot 1`] = `"a"`;

exports[`Process Tree Testsuite '/tau' conditional success tests AST matches snapshot 1`] = `"/tau"`;

exports[`Process Tree Testsuite '/rightarrow(a)' conditional success tests AST matches snapshot 1`] = `
{
  "operator": "/rightarrow",
  "subTrees": [
    "a",
  ],
}
`;

exports[`Process Tree Testsuite '/rightarrow(/tau)' conditional success tests AST matches snapshot 1`] = `
{
  "operator": "/rightarrow",
  "subTrees": [
    "/tau",
  ],
}
`;

exports[`Process Tree Testsuite '/rightarrow(a,/circlearrowleft(/rightarrow(/wedge(/times(b, c), d), e ), f ), /times(g, h))' conditional success tests AST matches snapshot 1`] = `
{
  "operator": "/rightarrow",
  "subTrees": [
    "a",
    {
      "operator": "/circlearrowleft",
      "subTrees": [
        {
          "operator": "/rightarrow",
          "subTrees": [
            {
              "operator": "/wedge",
              "subTrees": [
                {
                  "operator": "/times",
                  "subTrees": [
                    "b",
                    "c",
                  ],
                },
                "d",
              ],
            },
            "e",
          ],
        },
        "f",
      ],
    },
    {
      "operator": "/times",
      "subTrees": [
        "g",
        "h",
      ],
    },
  ],
}
`;

exports[`Process Tree Testsuite '' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "'/rightarrow'",
    "'/tau'",
    "action name",
  ],
  "pos": {
    "column": 1,
    "index": 0,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '1' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "'/rightarrow'",
    "'/tau'",
    "action name",
  ],
  "pos": {
    "column": 1,
    "index": 0,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '/rightarrow' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "'('",
  ],
  "pos": {
    "column": 12,
    "index": 11,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '/rightarrow(' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "')'",
    "'/circlearrowleft'",
    "'/rightarrow'",
    "'/tau'",
    "'/times'",
    "'/wedge'",
    "action name",
  ],
  "pos": {
    "column": 13,
    "index": 12,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '/rightarrow)' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "'('",
  ],
  "pos": {
    "column": 12,
    "index": 11,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '/rightarrow(a' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "')'",
    "','",
  ],
  "pos": {
    "column": 14,
    "index": 13,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '/rightarrowa)' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "'('",
  ],
  "pos": {
    "column": 12,
    "index": 11,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '/wedge(a)' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "'/rightarrow'",
    "'/tau'",
    "action name",
  ],
  "pos": {
    "column": 1,
    "index": 0,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '/circlearrowleft(a)' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "'/rightarrow'",
    "'/tau'",
    "action name",
  ],
  "pos": {
    "column": 1,
    "index": 0,
    "line": 1,
  },
}
`;

exports[`Process Tree Testsuite '/times(a)' conditional failure tests Error matches snapshot 1`] = `
{
  "expected": [
    "'/rightarrow'",
    "'/tau'",
    "action name",
  ],
  "pos": {
    "column": 1,
    "index": 0,
    "line": 1,
  },
}
`;
